File: Dart/lib/Problem2.dart
import 'package:Dart/Problem.dart' as Problem;

class Problem2 implements Problem.Problem {

  List<int> list;
  List<int> exp;


  Problem2(List<int> list, List<int> exp){
    this.list = list;
    this.exp = exp;
  }


  @override
  bool Calculate() {
    var results = new List(this.list.length);
    for (int i = 0; i<this.list.length; i++){

      int total = 1;
      for (int j= 0; j<this.list.length; j++){

        if (j != i){

          total = total*this.list[j];

        }


      }
      results[i] = total;

    }

    return true;

  }
}

File: PHP/Problem2.php
<?php
/**
 * Created by PhpStorm.
 * User: Animc
 * Date: 29-07-2018
 * Time: 00:11
 */

class Problem2 implements Problem
{

    public $list;
    public $exp;

    function __construct($list, $exp)
    {
        $this->list = $list;
        $this->exp = $exp;
    }

    public function Calculate()
    {
        $array = array();
        $l = count($this->list);
        for ($x = 0; $x < $l; $x++){
            $total = 1;
            for ($j = 0; $j < count($this->list); $j++){
                if ($j != $x){
                    $total = $total * $this->list[$j];
                }
            }
            array_push($array, $total);
        }
        return true;
    }
}